### ISSUE TITLE
Invalid json parsed without errors when passed via -var-file

### ISSUE THREAD

**Issue by keymon** [hashicorp/terraform#14673]
### Terraform Version

```
$ ~/local/terraform-0.9.5/terraform -v
Terraform v0.9.5
```
0.8.x also affected

### Affected Resource(s)

Command line parsing `-var-file`

### Terraform Configuration Files
Given this HCL:
```hcl
variable "mymap" {
  type        = "map"
}

output "mymap" {
  value="${var.mymap}"
}
```
and this .json:
```json
{
"mymap": {
    "key1": "1",
    "key2": "2"
    "key_after_missing_comma": "foo"
rubish
rubish
rubish
rubish
rubish everywhere
}
}
```

### Debug Output
https://gist.github.com/keymon/286653faefd318a5c8912f03250e80e5

### Expected Behavior

Error parsion json

### Actual Behavior

Works up to key2 :laughing: 
```
$ TF_LOG=TRACE TF_LOG_PATH=log ~/local/terraform-0.9.5/terraform apply -var-file spurious.json 

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

mymap = {
  key1 = 1
  key2 = 2
}

```

### Steps to Reproduce

Get the files above and:

`~/local/terraform-0.9.5/terraform apply -var-file spurious.json`

**Comment by Jonty** [comment]
@keymon ❤️

**Comment by apparentlymart** [comment]
Hi all! I just gave the original example a try on Terraform v0.12.0-alpha1, aside from some slightly different filenames:

```
$ terraform apply -var-file=invalid.tfvars.json

Error: Missing attribute separator comma

  on invalid.tfvars.json line 5:
   5:         "key_after_missing_comma": "foo"

A comma must appear between each property definition in an object.
```

The new JSON parser is strict about the input being valid JSON, and produces an error message as expected. Therefore I'm going to close this out. The fix is already merged in master and will be included in Terraform v0.12.0 final once released.

Thanks for reporting this, and sorry for the delay in resolving it.
